docs/Datalayer_for_GTM.md

```diff
@@ -0,0 +1,109 @@
+
+First, register the listener in `<MODULE>.services.yml`
+
+```yml
+  MODUILE_datalayer_alter_subscriber:
+    class: '\Drupal\<MODULE>\EventSubscriber\DatalayerAlterSubscriber'
+    tags:
+      - { name: 'event_subscriber' }
+
+```
+
+You'll need a file in your `src` folder in the `EventSubscriber` directory. For the service above, the file name is `DatalayerAlterSubscriber.php` so that Drupal will autoload it.
+
+```php
+<?php
+
+namespace Drupal\<MODULE>\EventSubscriber;
+
+use Drupal\node\Entity\Node;
+use Drupal\usa_twig_vars\Event\DatalayerAlterEvent;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+
+/**
+ * Add benefit category info to datalayer.
+ */
+class DatalayerAlterSubscriber implements EventSubscriberInterface {
+
+  /**
+   * {@inheritDoc}
+   */
+  public static function getSubscribedEvents() {
+    return [
+      DatalayerAlterEvent::EVENT_NAME => 'onDatalayerAlter',
+    ];
+  }
+
+  /**
+   * Adds category information to the datalayer.
+   */
+  public function onDatalayerAlter(DatalayerAlterEvent $event): void {
+    $node = \Drupal::routeMatch()->getParameter('node');
+    $event->datalayer['hasBenefitCategory'] = FALSE;
+
+    // make any changes need to $event->datalayer array
+    $event->datalayer['foo'] = 'bar';
+  }
+
+}
+```
+
+## Twig Templates
+
+The `html.html.twig` template outputs the payload as JSON. There are a number of flags required to render properly, the line should look like the line below. I've wrapped the line here for readability and added comments
+
+```php
+{{ datalayer |
+    // don't render accented characters as unicode entities
+    json_encode(constant('JSON_UNESCAPED_UNICODE')
+        // don't escape slashes in taxonomy URL and elsewhere
+        b-or constant('JSON_UNESCAPED_SLASHES')
+        b-or constant('JSON_PRETTY_PRINT'))
+    | raw }}
+```
+
+## Testing Changes
+
+You can use `scripts/drush/veryify-pagelayer-data.php` to validate that your changes don't change the datalayer output. This is more useful for refactoring the event subscribers or ensuring that additions to the data don't introduce unintended changes. Using the script takes some preparation:
+
+1. *Create Reference Output* The script compares Drupal outputs against a "known-good" data set, the published page CSV file created by a tome export. To prepare this file, you should export your local site using the code from the `dev` branch.
+2. *Add to the datalayer* In a new branch, create an event listener to make your changes.
+3. *Run verify script* use `drush php:script scripts/drush/verify-pagelayer-data.php`. It will fetch pages from your local site and compare the output of your datalayer to the expected values in the CSV file.
+
+The verify script takes two optional options, specified after `--`. Use `--base` if your local hostname is not reachable via `http://localhost/`. Ise the `--sample` flag for a quicker run that doesn't check every URL in the CSV file.
```

Comment:
It looks like you eliminated the `--base` option and added `--file` since you wrote this (and the notes on the this PR).